packages/@aws-cdk/aws-stepfunctions-tasks/lib/sagemaker-create-endpoint-config-task.ts

```diff
@@ -0,0 +1,152 @@
+            ...(this.renderProductionVariants(this.props.productionVariants)),
+            ...(this.renderDataCaptureConfig(this.props.dataCaptureConfig)),
+            ...(this.renderKmsKeyId(this.props.kmsKeyId)),
+            ...(this.renderTags(this.props.tags)),
+        };
+    }
+
+    private renderDataCaptureConfig(config: DataCaptureConfig | undefined): {[key: string]: any} {
+        return (config) ? { DataCaptureConfig: {
+            DestinationS3Uri: config.destinationS3Uri,
+            InitialSamplingPercentage: config.initialSamplingPercentage,
+            ...(config.kmsKeyId) ? {KmsKeyId: config.kmsKeyId} : {},
+            ...(config.enableCapture) ? {EnableCapture: config.enableCapture} : {},
+            ...(config.captureOptions) ? { CaptureOptions: config.captureOptions.map(
+                captureOption => ({
+                    CaptureMode: captureOption.captureMode
+                })
+            )} : {},
+            ...(config.captureContentTypeHeader) ? {
+                CaptureContentTypeHeader: {
+                    CsvContentTypes: config.captureContentTypeHeader.csvContentTypes,
+                    JsonContentTypes: config.captureContentTypeHeader.jsonContentTypes
+                }
+            } : {}
+        } } : {};
+    }
+
+    private renderKmsKeyId(config: string | undefined): {[key: string]: any} {
+        return (config) ? {KmsKeyId: config} : {};
+    }
+
+    private renderProductionVariants(configs: ProductionVariants[]): {[key: string]: any} {
+        return (configs) ? { ProductionVariants: configs.map(config => ({
+                ...(config.acceleratorType) ? { AcceleratorType: config.acceleratorType } : {},
+                ...(config.initialInstanceCount) ? { InitialInstanceCount: config.initialInstanceCount } : {},
+                ...(config.initialVariantWeight) ? { InitialVariantWeight: config.initialVariantWeight } : {},
+                ...(config.instanceType) ? { InstanceType: 'ml.' + config.instanceType.toString() } : {},
+                ...(config.modelName) ? { ModelName: config.modelName } : {},
+                ...(config.variantName) ? { VariantName: config.variantName } : {},
+            }))} : {};
+    }
+
+    private renderTags(tags: {[key: string]: any} | undefined): {[key: string]: any} {
+        return (tags) ? { Tags: Object.keys(tags).map(key => ({ Key: key, Value: tags[key] })) } : {};
+    }
+
+    private makePolicyStatements(task: sfn.Task): iam.PolicyStatement[] {
+        Stack.of(task);
+
+        // https://docs.aws.amazon.com/step-functions/latest/dg/sagemaker-iam.html
+        const policyStatements = [
+            new iam.PolicyStatement({
+                actions: ['sagemaker:CreateEndpointConfig'],
+                resources: ['*'],
+            }),
+            new iam.PolicyStatement({
+                actions: ['sagemaker:ListTags'],
+                resources: ['*']
+            }),
+        ];
```

Comment:
It is better to define the resources as follows
```
const stack = Stack.of(task);
const policyStatements = [
            new PolicyStatement({
                actions: ['sagemaker:CreateEndpointConfig'],
                resources: [
                    stack.formatArn({
                        service: 'sagemaker',
                        resource: 'endpoint-config',
                        resourceName: '*',
                    }),
                ],
            }),
            new PolicyStatement({
                actions: ['sagemaker:ListTags'],
                resources: [
                    stack.formatArn({
                        service: 'sagemaker',
                        resource: 'endpoint-config',
                        resourceName: '*',
                    }),
                ],
            }),
        ];
```

Reply:
+1 but I'll also add - given that we know the name of the endpoint config at this point, we should be able to add that to `resourceName` and avoid `*`.

Reply:
I was going to suggest specifying the resource name as well similar to:
```
            new PolicyStatement({
                actions: ['sagemaker:CreateEndpointConfig'],
                resources: [
                    stack.formatArn({
                        service: 'sagemaker',
                        resource: 'endpoint-config',
                        // resource name in sagemaker is lowercase
                        resourceName: Data.isJsonPathString(this.props.endpointConfigName) ? '*' : `${this.props.endpointConfigName.toLowerCase()}`,
                    }),
                ],
            }),
```
Please note that it is essential to convert resource name to lower case as Sagemaker arns lowercase the name by default(for any resource be it training job, endpoint, ,model etc). I found this case undocumented